### ISSUE TITLE
Honey Select 2 Not Start

### ISSUE THREAD

**Issue by MichaelEvil** [BepInEx/BepInEx#160]
**Describe the bug**

Downloaded HS2 R5 and the game cannot start.

**To Reproduce**

Before I had R4 and it worked well until last Friday. That day the game (main game, vr, studiov2) can not start even the console not pop out and no output_log.txt, tried Administrator to run and no effect.
I renamed BepInEx folder to _BepInEx, the game starts correctly, but when I click girl character customization, the game goes into a full white screen with normal background music playing.
Discord guys suggest I get a new copy so I downloaded R5 but still the same situation.

**Expected behavior**

The game starts correctly with BepInEx on.

**Screenshots and logs**

Nothing helpful I think.

**Desktop (please complete the following information):**
Computer Specs: R5-3600X RTX2070S
System Specs: Windows 10 20H2 Enterprise
 - Game and game version [e.g. COM3D2 1.15]: HS2 R5 FINAL
 - BepInEx version [e.g. 4.0]: 5.3

**Additional context**

I am pretty sure that between the last time R4 runs normally and this time on Friday it totally fails, nothing changed in the game folder because I lock the hard drive with BitLocker and I would unlock it only when playing games in it.
The R5 installation file is just downloaded and it fails the same way so I think the game and BepInEx itself should not be the reason. The only change that occurred on my computer I think is some windows update...
Can't dig out more information. I am willing to help to do any test on my device. Quite curious about this problem.

**Comment by denikson** [comment]
Greetings!

From my personal experience with BitLocker and similar encryption tools like VeraCrypt, they usually apply some kind of caching or additional behaviour that can affect running Doorstop which is the component of BepInEx responsible for launching the mod loader itself.

Please send over the following logs as it was suggested in the issue template:
* `output_log.txt` from game's `_Data` folder
* Any logs that exist next to game EXE and the names of which start with `preloader_err`

Next, please try replacing `winhttp.dll` with the following version that generates logs:
[doorstop_verbose.zip](https://github.com/BepInEx/BepInEx/files/5654444/doorstop_verbose.zip)
(If I recall correctly, HS2 is x64 so I'm sending x64 version only)

Run the game, after which it should generate `doorstop_XXXX.log` file next to game executable (where `XXXX` is a sequence of characters). Please send the logs or report if they are not generated (which indicates problems with BitLocker itself).

Overall, I've had better luck with using VeraCrypt than BitLocker when it comes to games.

**Comment by MichaelEvil** [comment]
Thanks for your quick response! I'll have a try when I back home night (GMT-8)😘

**Comment by MichaelEvil** [comment]
@denikson I tried the methods you mentioned and found something interesting.
1. There was no log file named "output_log" or "preloader" in the game folder before I did step 2.
2. I tried your debug .dll file and the game started normally in Japanese. I click girl character customization and it goes white screen.
Some logs generated:
1) doorstop_d1324c6.log. It reported an error about TimeZone TimeZoneNotFoundException
```
Invoking method 000002641CFAC638
Error invoking code!
Error message: System.TimeZoneNotFoundException: Exception of type 'System.TimeZoneNotFoundException' was thrown.
  at System.TimeZoneInfo.FindSystemTimeZoneById (System.String id) [0x00039] in <7ba07f088431485bb722f3b3373e87ee>:0 
  at System.TimeZoneInfo.CreateLocal () [0x00053] in <7ba07f088431485bb722f3b3373e87ee>:0 
  at System.TimeZoneInfo.get_Local () [0x00009] in <7ba07f088431485bb722f3b3373e87ee>:0 
  at System.TimeZoneInfo.GetDateTimeNowUtcOffsetFromUtc (System.DateTime time, System.Boolean& isAmbiguousLocalDst) [0x00000] in <7ba07f088431485bb722f3b3373e87ee>:0 
  at System.DateTime.get_Now () [0x00008] in <7ba07f088431485bb722f3b3373e87ee>:0 
  at BepInEx.Preloader.Entrypoint.Main () [0x00000] in <f9454f44494642d6aa1808eb7ec9b41e>:0 
Done!
```
[doorstop_d1324c6.log](https://github.com/BepInEx/BepInEx/files/5659890/doorstop_d1324c6.log)


2) output_log.txt. Basically two types of error loop.
```
TimeZoneNotFoundException: Exception of type 'System.TimeZoneNotFoundException' was thrown.
  at System.TimeZoneInfo.FindSystemTimeZoneById (System.String id) [0x00039] in <7ba07f088431485bb722f3b3373e87ee>:0 
  at System.TimeZoneInfo.CreateLocal () [0x00053] in <7ba07f088431485bb722f3b3373e87ee>:0 
  at System.TimeZoneInfo.get_Local () [0x00009] in <7ba07f088431485bb722f3b3373e87ee>:0 
  at System.DateTime.ToLocalTime (System.Boolean throwOnOverflow) [0x00014] in <7ba07f088431485bb722f3b3373e87ee>:0 
  at System.DateTime.ToLocalTime () [0x00000] in <7ba07f088431485bb722f3b3373e87ee>:0 
  at System.DateTime.FromFileTime (System.Int64 fileTime) [0x00007] in <7ba07f088431485bb722f3b3373e87ee>:0 
  at System.IO.File.GetLastWriteTime (System.String path) [0x00027] in <7ba07f088431485bb722f3b3373e87ee>:0 
  at FolderAssist+<>c.<CreateFolderInfoExToArray>b__3_1 (System.String path) [0x00007] in <aebd3b6ac6534b75930264311ce5b75d>:0 
  at System.Linq.Enumerable+SelectArrayIterator`2[TSource,TResult].ToArray () [0x00012] in <fbb5ed17eb6e46c680000f8910ebb50c>:0 
  at System.Linq.Enumerable.ToArray[TSource] (System.Collections.Generic.IEnumerable`1[T] source) [0x0001f] in <fbb5ed17eb6e46c680000f8910ebb50c>:0 
  at FolderAssist.CreateFolderInfoExToArray (System.String folder, System.String[] searchPattern) [0x00063] in <aebd3b6ac6534b75930264311ce5b75d>:0 
  at FolderAssist.CreateFolderInfoEx (System.String folder, System.String[] searchPattern, System.Boolean clear) [0x0000e] in <aebd3b6ac6534b75930264311ce5b75d>:0 
  at SaveFrameAssist.ChangeSaveFrameBack (System.Byte changeNo, System.Boolean listUpdate) [0x0003a] in <aebd3b6ac6534b75930264311ce5b75d>:0 
  at SaveFrameAssist.Initialize () [0x00000] in <aebd3b6ac6534b75930264311ce5b75d>:0 
  at CharaCustom.CustomControl.Initialize (System.Byte _sex, System.Boolean _new, System.String _nextScene, System.String _editCharaFileName, System.Boolean _isConcierge, System.Boolean _isPlayer) [0x00051] in <aebd3b6ac6534b75930264311ce5b75d>:0 
  at CharaCustom.CharaCustom+<Start>d__15.MoveNext () [0x000a9] in <aebd3b6ac6534b75930264311ce5b75d>:0 
  at UnityEngine.SetupCoroutine.InvokeMoveNext (System.Collections.IEnumerator enumerator, System.IntPtr returnValueAddress) [0x00027] in <a43c47796b1b4dc78509dc74aec2356b>:0 
 
(Filename: <7ba07f088431485bb722f3b3373e87ee> Line: 0)

NullReferenceException: Object reference not set to an instance of an object
  at CharaCustom.CvsBase.get_makeup () [0x00006] in <aebd3b6ac6534b75930264311ce5b75d>:0 
  at CharaCustom.CvsF_MakeupLip+<Start>d__8.MoveNext () [0x0006a] in <aebd3b6ac6534b75930264311ce5b75d>:0 
  at UnityEngine.SetupCoroutine.InvokeMoveNext (System.Collections.IEnumerator enumerator, System.IntPtr returnValueAddress) [0x00027] in <a43c47796b1b4dc78509dc74aec2356b>:0 
```
[output_log.txt](https://github.com/BepInEx/BepInEx/files/5659901/output_log.txt)


Since no any preloader-*.log file found, I searched the code and found this in BepInEx.Preloader.Unity/DoorstopEntrypoint.cs:
```
string silentExceptionLog = $"preloader_{DateTime.Now:yyyyMMdd_HHmmss_fff}.log";
```
Maybe error starts here?
Hope the things above can help.

**Comment by denikson** [comment]
What locale are you on? I'd suggest trying to set system's locale to either Japanese or English.
Alternatively, you can try using [Locale Emulator](https://xupefei.github.io/Locale-Emulator/) to run the game with the locale emulation.

**Comment by MichaelEvil** [comment]
Oh my... Thanks, my friend!
I changed my locale from GMT8 to GMT9 then **all the problems solved**. Then I changed it back and the game still works perfectly. Excellent! Seems the game or something else needs a "first-time" verification?
That's great but I am still quite confused about this because I knew HS for years and locale never become a problem.
By the way, LE does not support x64 programs😨. Though I seem not to need this in a short time, is there any alternative?
